packages/svelte-vscode/src/sveltekit/utils.ts

```diff
@@ -26,39 +26,47 @@ export async function findFile(searchPath: string, fileName: string) {
     }
 }
 
-export async function checkProjectType(path: string): Promise<ProjectType> {
+export async function checkProjectKind(path: string): Promise<GenerateConfig['kind']> {
     const tsconfig = await findFile(path, 'tsconfig.json');
     const jsconfig = await findFile(path, 'jsconfig.json');
-    const svelteVersion = await getSvelteVersionFromPackageJson();
-    const isSv5Plus = isSvelte5Plus(svelteVersion);
-    const isTs = !!tsconfig && (!jsconfig || tsconfig.length >= jsconfig.length);
-    if (isTs) {
+
+    const svelteVersion = await getVersionFromPackageJson('svelte');
+    const withRunes = svelteVersion ? versionAtLeast(svelteVersion, 5) : true;
+
+    const svelteKitVersion = await getVersionFromPackageJson('@sveltejs/kit');
+    let withProps = svelteKitVersion ? versionAtLeast(svelteKitVersion, 2, 16) : true;
+
+    const withTs = !!tsconfig && (!jsconfig || tsconfig.length >= jsconfig.length);
+    let withSatisfies = false;
+    if (withTs) {
         try {
             const packageJSONPath = require.resolve('typescript/package.json', {
                 paths: [tsconfig]
             });
             const { version } = require(packageJSONPath);
-            const [major, minor] = version.split('.');
-            if ((Number(major) === 4 && Number(minor) >= 9) || Number(major) > 4) {
-                return isSv5Plus ? ProjectType.TS_SATISFIES_SV5 : ProjectType.TS_SATISFIES;
-            } else {
-                return isSv5Plus ? ProjectType.TS_SV5 : ProjectType.TS;
-            }
+            withSatisfies = version ? versionAtLeast(version, 4, 9) : true;
         } catch (e) {
-            return isSv5Plus ? ProjectType.TS_SV5 : ProjectType.TS;
+            withSatisfies = true;
```

Comment:
If for some reason we don't find typescript Version, fallback to withSatisfies.
What do you think ?